PittAPI/laundry.py

```diff
@@ -49,92 +36,85 @@ def get_status_simple(building_name):
 
-    # Get the weird laundry data
-    headers = {"Cookie": "PHPSESSID={}".format(cookie)}
-    response = requests.get("http://www.laundryview.com/dynamicRoomData.php?location={}".format(
-        location_dict[building_name]), headers=headers).text
-    resp_split = response.split('&')[3:]
+              A dict of machine metadata if machine name
+              is explicity passed
 
-    cleaned_resp = []
-    for status_string in resp_split:
-        if "machine" not in status_string:
+    :param building_name: (String) one of these:
+        -> BRACKENRIDGE
+        -> HOLLAND
+        -> LOTHROP
+        -> MCCORMICK
+        -> SUTH_EAST
+        -> SUTH_WEST
+    :param machine: (String) specific machine to query
+                    eg: machine='BA02'
+    """
+    building_name = building_name.upper()
+    machines = []
+
+    url = 'http://m.laundryview.com/submitFunctions.php?monitor=true&lr={}'.\
+            format(location_lookup[building_name])
+    response = requests.get(url)
+    laundry_soup = BeautifulSoup(response.text, 'lxml')
+    is_washer = False
+
+    for li in laundry_soup.findAll('li'):
+        if 'id' in li.attrs:
+            is_washer = True if li.attrs['id'] == 'washer' \
+                            else False
             continue
-        machine_name = status_string[:status_string.index('=')].replace('Status', '')
-        status_string = status_string[status_string.index('=') + 1:].strip()
 
-        machine_split = status_string.split("\n")
-        machine_split[0] += machine_name
+        machine_id = int(li.find('a').attrs['id'])
+        machine_status = li.find('p').text
+        machine_name = li.text.split(machine_status)[0].\
+                           encode('ascii', 'ignore')
+        machine_type = 'washer' if is_washer else 'dryer'
 
-        try:
-            machine_split[1] += machine_name
-        except IndexError:
-            pass
-
-        machine_split = [x.split(':') for x in machine_split]
-        cleaned_resp.append(machine_split[0])
-        try:
-            cleaned_resp.append(machine_split[1])
-        except IndexError:
-            pass
-
-    cleaned_resp = [x for x in cleaned_resp if len(x) == 10]
-
-    di = []
-    for machine in cleaned_resp:
-        time_left = -1
-        machine_name = "{}_{}".format(machine[9], machine[3])
-        machine_status = ""
-
-        if machine[0] is '1':
-            machine_status = u'Free'
-        else:
-            if machine[6] is '':
-                machine_status = u'Out of service'
-            else:
-                machine_status = u'In use'
-
-        if machine_status is u'In use':
-            time_left = int(machine[1])
-        else:
-            time_left = -1 if machine[6] is '' else machine[6]
-        di.append({
+        machines.append({
             'machine_name': machine_name,
+            'machine_type': machine_type,
             'machine_status': machine_status,
-            'time_left': time_left
```

Comment:
`time_left` is critical. Do not take this away.

Reply:
Well we have a `machine_status` that indicates the number of minutes left when it is being used.  It is `Avail` if idle (and online). Not sure about the significance of a separate field.

Reply:
`time_left` is the number of minutes until it's finished

Reply:
Having a string showing the whole thing is not as useful as having an explicit field with an integer, which is why that field is there. Also, this would introduce breaking API changes.